docs/pages/database-access/enroll-aws-databases/redshift-serverless.mdx

```diff
@@ -0,0 +1,363 @@
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "Statement1",
+            "Effect": "Allow",
+            "Principal": {
+                "AWS": "arn:aws:iam::(=aws.aws_access_key=):role/teleport-redshift-serverless-node",
+                "Service": "ec2.amazonaws.com"
+            },
+            "Action": "sts:AssumeRole"
+        }
+    ]
+}
+```
+
+Remember to replace the example AWS account ID.
+
+Create and apply a permission policy to allow access to Redshift Serverless.
+This policy can allow access to all instances:
+
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": "redshift-serverless:GetCredentials",
+            "Resource": "*"
+        }
+    ]
+}
+```
+
+Or you can restrict the `Resource` line to a specific Redshift Serverless workgroup:
+
+```json
+{
+...
+      "Resource": "arn:aws:redshift-serverless:us-west-2:1234567890:workgroup/some-workgroup-id"
+...
+}
+```
+
+## Step 3/5. Create a Teleport role for Redshift Serverless access
+
+On your workstation logged in to your Teleport cluster with `tsh`, define a new
+role to provide access to Redshift Serverless. Our example file is
+`redshift-role.yaml`:
+
+```yaml
+version: v5
+kind: role
+metadata:
+  name: redshift-serverless-access
+spec:
+  allow:
+    db_labels:
+      '*': '*'
+    db_names:
+    - dev
+    db_users:
+    - 'teleport-redshift-serverless-access'
+```
+
+- The value of `db_users` corresponds to the IAM role created in the previous step.
+  You can provide either the role name or the full AWS ARN of the IAM role.
+- The value(s) for `db_names` will depend on your Redshift Serverless configuration,
+  but `dev` is the default name applied by AWS. You can also provide `*` to grant
+  access to all instances.
+
+Save this file and apply it to your Teleport cluster:
+
+```code
+$ tctl create -f redshift-role.yaml
+role 'redshift-serverless-access' has been created
+```
+
+(\!docs/pages/includes/add-role-to-user.mdx role="redshift-serverless-access" \!)
+
+## Step 4/5. Install and start the Teleport Database Service
+
+(!docs/pages/includes/database-access/token.mdx!)
+
+(!docs/pages/includes/database-access/alternative-methods-join.mdx!)
+
+Install Teleport on the host where you will run the Teleport Database Service:
+
+(!docs/pages/includes/install-linux.mdx!)
+
+On the same host, create a file at `/etc/teleport.yaml` with the following
+content.
+
+```yaml
+version: v3
+teleport:
+  data_dir: /var/lib/teleport
+  proxy_server: teleport.example.com:3080
+  auth_token: "/tmp/token"
+db_service:
+  enabled: "yes"
+  # This section contains definitions of all databases proxied by this
+  # service, can contain multiple items.
+  databases:
+  - name: "redshift-serverless"
+    uri: REDSHIFT_SERVERLESS_URI
+    protocol: postgres
+    aws:
+      redshift_serverless:
+        workgroup_name: WORKGROUP_NAME
+        endpoint_name: ENDPOINT_NAME
```

Comment:
`WORKGROUP_NAME` and `ENDPOINT_NAME` are both optional (as they can be parsed from URI as well). `ENDPOINT_NAME` is only for VPC endpoints, so regular workgroups don't need it.

Alternatively, we can also use `teleport db configure create` to generate the db config, with protocol and URI.